js/ui/viewSelector.js

```diff
@@ -399,7 +399,11 @@ const ViewsDisplay = new Lang.Class({
     },
 
     _updateSpinner: function() {
-        this.entry.setSpinning(this._searchResults.searchInProgress);
+        // Make sure we never set the spinner on when there's nothing to search,
+        // regardless of the reported current state, as it can be out of date.
+        let searchTerms = this.entry.text.trim();
+        let spinning =  searchTerms.length > 0 ? this._searchResults.searchInProgress : false;
```

Comment:
Not a big deal, but perhaps slightly cleaner:
`let spinning = searchTerms.length > 0 && this._searchResults.searchInProgress;`
(also remove the extra space after the `=`)